dbt/adapters/dremio/connections.py

```diff
@@ -230,6 +231,102 @@ def create_catalog(self, relation):
             self._create_folders(database, schema, rest_client)
         return
 
+    def create_reflection(self, name: str, type: str, anchor: DremioRelation, display: List[str], dimensions: List[str],
+                          date_dimensions: List[str], measures: List[str],
+                          computations: List[str], partition_by: List[str], partition_transform: List[str],
+                          partition_method: str, distribute_by: List[str], localsort_by: List[str],
+                          arrow_cache: bool) -> None:
+        thread_connection = self.get_thread_connection()
+        connection = self.open(thread_connection)
+        rest_client = connection.handle.get_client()
+
+        database = anchor.database
+        schema = anchor.schema
+        path = self._create_path_list(database, schema)
+        identifier = anchor.identifier
+
+        path.append(identifier)
+
+        catalog_info = rest_client.get_catalog_item(
+            catalog_id=None,
+            catalog_path=path,
+        )
+
+        dataset_id = catalog_info.get("id")
+
+        payload = {
```

Comment:
Let's start creating entities for requests with fields inside it.
And make them a part of the rest client.